website/src/documentation/programming-guide.md

```diff
@@ -3121,3 +3121,515 @@ public class MyMetricsDoFn extends DoFn<Integer, Integer> {
   }
 }
 ```  
+
+## 10. State and Timers {#state-and-timers}
+Beam's windowing and triggering facilities provide a powerful abstraction for grouping and aggregating unbounded input
+data based on timestamps. However there are aggregation use cases for which Beam's windows and triggers are not the best
```

Comment:
However there are aggregation use cases for which developers may require a higher degree of control...

Reply:
I like it. done